prompt/prompt-model/src/commonMain/kotlin/ai/koog/prompt/message/Message.kt

```diff
@@ -150,8 +164,60 @@ public sealed interface Message {
      */
     @Serializable
     public data class System(
-        override val content: String
+        override val content: String,
+        override val metadata: RequestMetadata = RequestMetadata()
     ) : Request {
         override val role: Role = Role.System
     }
 }
+
+/**
+ * Metadata associated with a message in a chat system.
+ *
+ * @property timestamp The timestamp of when the message is created, measured in milliseconds
+ * since the Unix epoch. Defaults to the current system time.
+ */
+@Serializable
+public sealed interface MessageMetadata {
+    /**
+     * Represents the timestamp of a message
+     *
+     * This property indicates the precise time when a message was created. It defaults
+     * to the current system time if not explicitly set.
+     */
+    public val timestamp: Instant
+}
+
+/**
+ * Represents [MessageMetadata] specific to a request within the system.
+ *
+ * This class is an implementation of the [MessageMetadata] interface and provides
+ * timestamp information for a request.
+ *
+ * @property timestamp The time at which the request metadata was created.
+ * Defaults to the current system time if not provided.
+ */
+@Serializable
+public data class RequestMetadata(
+    override val timestamp: Instant = Clock.System.now()
+) : MessageMetadata
+
+/**
+ * Represents [MessageMetadata] for a response message.
+ *
+ * @property tokensCount The total count of tokens used by the time this response message is returned by the LLM.
+ *                       This includes the tokens for the entire chat history up to and including this message,
+ *                       not just the tokens for the current message.
+ * @constructor Creates an instance of `ResponseMessageMetadata` with the specified total token count.
+ *              Optionally, a custom timestamp can be provided; otherwise, the current system time is used.
+ *
+ * Example:
+ * - Message 1: "Hello" (3 tokens) → tokensCount = 3
+ * - Message 2: "How are you?" (4 tokens) → tokensCount = 3 + 4 = 7
+ * - Message 3: "I am fine, thank you." (6 tokens) → tokensCount = 7 + 6 = 13
+ */
+@Serializable
+public data class ResponseMetadata(
+    public val tokensCount: Int? = null,
+    override val timestamp: Instant = Clock.System.now()
```

Comment:
I would not introduce Clock.System.now() as default value for timestamp, because it's kinda error prone. For me cleaner way here - to require calling side to stamp their requests

Reply:
>Then, whenever one wants to create any Message even in tests -- they would have to manually create ResponseMetadata and pass the timestamp

I don't think it's a problem. That's DTO after all, it shouldn't timestamp itself. Moreover, users might want to control what stamps they want to put on the messages.